### PR TITLE
fix: Use the new Codecov uploader

### PR DESCRIPTION
**Pull request by thomasrockhu-codecov**
Uses the new Codecov uploader as documented [here](https://docs.codecov.com/docs/codecov-uploader)

### CODE REVIEW

**Review by hseligson1 — APPROVED** [review]

**Review by hseligson1 — APPROVED** [review]
Updating step for Codecov changes.

File: incubating/codecov-reporter/step.yaml [hunk 1]
```diff
@@ -57,13 +57,18 @@ spec:
         "additionalProperties": false,
         "patterns": [],
         "required": [
-            "codecov_integration"
+            "codecov_integration",
+            "os"
```
Comment (pasha-codefresh) [inline incubating/codecov-reporter/step.yaml:61]: It is breaking change, will cause problem with existing customers, I think we should add at least some default value

File: incubating/codecov-reporter/step.yaml [hunk 2]
```diff
@@ -57,13 +57,18 @@ spec:
...
         ],
         "properties": {
             "codecov_integration": {
                 "type": "string",
                 "description": "Codecov integration name"
             },
+            "os": {
+                "type": "string",
+                "description": "OS being used ( alpline | linux | macos | windows )"
```
Comment (pasha-codefresh) [inline incubating/codecov-reporter/step.yaml:70]: here is typo
alpline

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

File: incubating/codecov-reporter/step.yaml [hunk 3]
```diff
@@ -88,7 +93,13 @@ spec:
...
-        - codecov -t ${{CODECOV_API_KEY}} | tee codecov.log
-        - cf_export second_CF_OUTPUT_URL=$(grep -A0 'View report at:' codecov.log | awk 'BEGIN{FS="at:\ ";}{print $2}')
+        - curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import # One-time step
+        - curl -Os "https://uploader.codecov.io/latest/${{os}}/codecov"
+        - curl -Os "https://uploader.codecov.io/latest/${{os}}/codecov.SHA256SUM"
+        - curl -Os "https://uploader.codecov.io/latest/${{os}}/codecov.SHA256SUM.sig"
+        - gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
+        - shasum -a 256 -c codecov.SHA256SUM
+        - chmod +x codecov
+        - ./codecov -t ${{CODECOV_API_KEY}} -Q codefresh-2.0.0 | tee codecov.log
```
Comment (pasha-codefresh) [inline incubating/codecov-reporter/step.yaml:103]: when i execute it , ./codecov can`t be found

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

**Comment by pasha-codefresh** [comment]
@thomasrockhu-codecov Thank you for your contribution, it is merged 
https://github.com/codefresh-io/steps/pull/399 
I created another Pr because I did changes

**Pull request closed without merging**